Source/WebKit/PlatformGTK.cmake

```diff
@@ -544,6 +551,34 @@ if (ENABLE_MEDIA_STREAM)
     )
 endif ()
 
+foreach (TEMPLATE ${WebKit2GTK_HEADER_TEMPLATES})
+    get_filename_component(HEADER ${TEMPLATE} NAME_WLE)
+
+    add_custom_command(
+        OUTPUT ${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER}
+        DEPENDS ${TEMPLATE} unifdef
+        COMMAND ${CMAKE_BINARY_DIR}/bin/unifdef -D "PLATFORM(GTK)=1" -D "PLATFORM(WPE)=0" -x2 -o ${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER} ${TEMPLATE}
+        VERBATIM
+    )
+
+    list(APPEND WebKit2GTK_INSTALLED_HEADERS "${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER}")
+endforeach ()
+list(APPEND WebKit_SOURCES ${WebKit2GTK_INSTALLED_HEADERS})
+
+foreach (TEMPLATE ${WebKit2WebExtension_HEADER_TEMPLATES})
+    get_filename_component(HEADER ${TEMPLATE} NAME_WLE)
+
+    add_custom_command(
+        OUTPUT ${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER}
+        DEPENDS ${TEMPLATE} unifdef
+        COMMAND ${CMAKE_BINARY_DIR}/bin/unifdef -D "PLATFORM(GTK)=1" -D "PLATFORM(WPE)=0" -x2 -o ${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER} ${TEMPLATE}
+        VERBATIM
+    )
+
+    list(APPEND WebKit2WebExtension_INSTALLED_HEADERS "${WebKit2Gtk_DERIVED_SOURCES_DIR}/webkit/${HEADER}")
+endforeach ()
+list(APPEND WebKit_SOURCES ${WebKit2WebExtension_INSTALLED_HEADERS})
```

Comment:
Could this be converted into a macro to avoid the code duplication?

Reply:
Probably: it really needs to be, because otherwise it will become a big mess as soon as we introduce any more build configurations. I was planning to attempt that in a follow-up patch, but I'll try now.

Reply:
Fixed, now uses a macro.